frontend/appflowy_flutter/lib/workspace/presentation/settings/shared/document_color_setting_button.dart

```diff
@@ -241,3 +263,107 @@ String? validateOpacityValue(String? value) {
   }
   return null;
 }
+
+void _showColorPickerDialog({
+  required BuildContext context,
+  String? title,
+  required Color currentColor,
+  required void Function(Color) updateColor,
+}) {
+  const kColorCircleWidth = 46.0;
+  const kColorCircleHeight = 46.0;
+  const kColorCircleRadius = 23.0;
+  const kColorOpacityThumbRadius = 23.0;
+  const kDialogButtonPaddingHorizontal = 24.0;
+  const kDialogButtonPaddingVertical = 12.0;
+  const kColorsColumnSpacing = 3.0;
+  final style = Theme.of(context);
+  Color selectedColor = currentColor;
+
+  void updated(Color color) {
+    updateColor(color);
+    Navigator.of(context).pop();
+  }
+
+  showDialog(
+    context: context,
+    barrierColor: const Color.fromARGB(128, 0, 0, 0),
+    builder: (context) {
+      return AlertDialog(
+        icon: const Icon(Icons.palette),
+        title: Text(
+          title ??
+              LocaleKeys.settings_appearance_documentSettings_pickColor.tr(),
```

Comment:
It would be better to subtract a widget to handle the logic instead of putting all of them in a function.